sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/implementation/util/TimeUtils.java

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.changefeed.implementation.util;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoUnit;
+
+public class TimeUtils {
+
+    /**
+     * Converts a path to an OffsetDateTime.
+     * <p>For example,
+     * <p>Segment path : idx/segments/1601/01/01/0000/meta.json
+     * <p>OffsetDateTime : year - 1601, month - 01, day - 01, hour - 00, minute - 00
+     * <p>OR
+     * <p>Year path : idx/segments/1601/
+     * <p>OffsetDateTime : year - 1601, month - 00, day - 00, hour - 00, minute - 00
+     *
+     * @param path The path to convert.
+     * @return The time associated with the path.
+     */
+    public static OffsetDateTime convertPathToTime(String path) {
+        if (path == null) {
+            return null;
+        }
+        String[] splitPath = path.split("/");
+
+        return OffsetDateTime.of(
+            Integer.parseInt(splitPath[2]), /* year */
+            splitPath.length < 4 ? 1 : Integer.parseInt(splitPath[3]), /* month */
+            splitPath.length < 5 ? 1 : Integer.parseInt(splitPath[4]), /* day */
+            splitPath.length < 6 ? 0 : Integer.parseInt(splitPath[5]) / 100, /* hour */
+            0, /* minute */
```

Comment:
Given a segment path `idx/segments/2020/05/28/1047/meta.json` why do we always set minute to `0`? Instead should we set it to the following

```java
splitPath.length < 6 ? 0 : Integer.parseInt(splitPath[5]) % 100 /* minute */
```

Reply:
In changefeed a segment is a representation of an hour, so I just set min and sec to 0, 0